File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int mini = prices[0], profit = 0, cost = 0, n = prices.size();
        for(int i=1; i<n; i++){
            cost = prices[i]-mini;
            profit = max(profit,cost);
            mini = min(mini,prices[i]);
        }
        return profit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let maxP = 0;
    let n = prices.length;
    let l = 0, r = 1;
    while(r<n){
        if(prices[l]<prices[r]){
            let profit = prices[r]-prices[l];
            maxP = Math.max(maxP,profit);
        }else{
            l=r;
        }
        r++;
    }
    return maxP;
};